lib/url.js

```diff
@@ -353,6 +353,13 @@ function urlFormat(obj) {
   // this way, you can call url_format() on strings
   // to clean up potentially wonky urls.
   if (typeof obj === 'string') obj = urlParse(obj);
+
+  if (typeof obj !== 'object')
```

Comment:
I did choose to use only `if` for readability, but yes, I should probably change it to `else if` for speed.
Also, I did prefer the explicit error message over one `else if`.